### PR TITLE
Rrojo/pci device updates

### PR DESCRIPTION
**Pull request by r-rojo**
Several improvements as well as sub-commands to pci_device script.

* Change the first positional argument to be either the pci address or a
vendor/device ID pair. In either case, this input is treated like a
filter for enumerating devices and whatever sub-command is given will
operate on the resulting list of devices returned from this enumeration.

* Add `aer` subcommand
  * This subcommand has it's own subcommands: `dump` or `mask` (with default
    being `dump`)
  * `dump` subcommand will print out the aer error counters as reported
     by the sysfs files for the device.
  * `mask` can either print out the current AER mask bits or set them
    * If `show` or `print` (or nothing) is given after the `mask`
    command, it will show the current mask bits for AER
    * If `all` is given after the `mask` command, it will mask all bits
      (by setting the values to 0xffffffff and 0xffffffff).
    * If `off` is given after the `mask` command, it will unmask all
    bits ( by setting the values to 0x0 and 0x0).
    * If two numbers are present after the `mask` command, those two
    numbers will be used to set the mask bits.

* Update documentation to reflect these changes.

### CODE REVIEW

File: doc/src/fpga_tools/pci_device/pci_device.md [hunk 1]
```diff
@@ -12,14 +12,40 @@ drivers.
 ## OPTIONS ##
 
 ### POSITIONAL ARGUMENTS ###
-    `device`
+    `device filter`
 
-    pcie address of device ([segment:]bus:device.function)
+    pcie address of a device or vendor/device ID pair.
```
Comment (tswhison) [inline doc/src/fpga_tools/pci_device/pci_device.md:17]: s,pcie,PCIe,g

File: doc/src/fpga_tools/pci_device/pci_device.md [hunk 2]
```diff
@@ -12,14 +12,40 @@ drivers.
...
+    while the vendor/device ID pair follows the format [vendor ID]:[device ID]
+    where at least one of these must be present.
 
-    `{remove,rescan,topology,unbind}`
+    `{aer,remove,rescan,topology,unbind}`
 
     action to perform on device
 
+    `aer`
+    Perform AER (Advandced Error Reporting) operations.
```
Comment (tswhison) [inline doc/src/fpga_tools/pci_device/pci_device.md:27]: s,Advandced,Advanced,

File: doc/src/fpga_tools/pci_device/pci_device.md [hunk 3]
```diff
@@ -12,14 +12,40 @@ drivers.
...
+    The aer action has its own sub-commands and listed below:
```
Comment (tswhison) [inline doc/src/fpga_tools/pci_device/pci_device.md:28]: s,and,which are,

File: doc/src/fpga_tools/pci_device/pci_device.md [hunk 4]
```diff
@@ -12,14 +12,40 @@ drivers.
...
+
+    * `dump` sub-command will print out the aer error counters as reported
```
Comment (tswhison) [inline doc/src/fpga_tools/pci_device/pci_device.md:30]: s,aer,AER,

File: doc/src/fpga_tools/pci_device/pci_device.md [hunk 5]
```diff
@@ -12,14 +12,40 @@ drivers.
...
+       by the sysfs files for the device.
+    * `mask` can either print out the current AER mask bits or set them
+      * If `show` or `print` (or nothing) is given after the `mask`
+        command, it will show the current mask bits for AER.
+	By default output will be written in stdout but can be written to an
+	output file if `-o|--output FILENAME` argument is given.
+      * If `all` is given after the `mask` command, it will mask all bits
+        (by setting the values to 0xffffffff and 0xffffffff).
+      * If `off` is given after the `mask` command, it will unmask all
```
Comment (tswhison) [inline doc/src/fpga_tools/pci_device/pci_device.md:39]: s,`mask`,`unmask`,

File: doc/src/fpga_tools/pci_device/pci_device.md [hunk 6]
```diff
@@ -12,14 +12,40 @@ drivers.
...
+        bits ( by setting the values to 0x0 and 0x0).
```
Comment (tswhison) [inline doc/src/fpga_tools/pci_device/pci_device.md:40]: s,( by,(by,

File: doc/src/fpga_tools/pci_device/pci_device.md (hunk 5 above)
Comment (r-rojo) [reply doc/src/fpga_tools/pci_device/pci_device.md:39]: This is correct

Comment (tswhison) [reply doc/src/fpga_tools/pci_device/pci_device.md:39]: my bad :)

**Review by tswhison — APPROVED** [review]

**Pull request merged**